pkg/syncer/k8scloudoperator/placement.go

```diff
@@ -321,6 +315,28 @@ func getSPForPVCPlacement(ctx context.Context, curPVC *v1.PersistentVolumeClaim,
 		return assignedSP, fmt.Errorf("fail to find a StoragePool passing all criteria")
 	}
 
+	xCapPending := 0
+	for _, pvcItem := range pvcList {
+		if pvcItem.Status.Phase == v1.ClaimPending {
+			spName, ok := pvcItem.Annotations[StoragePoolAnnotationKey]
+			if !ok {
+				continue
+			}
+
+			// update SP usage based on any unbound PVCs placed on this SP. These are still in pipeline and hence
+			// the usage of SP will not be updated yet. There is always a race where the usage is already updated but
+			// the PVC is not yet in bound state but we will rather be conservative and try the placement again later
+			capacity := pvcItem.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+			spRemoved := false
+			_, spRemoved, spList = updateSPCapacityUsage(spList, spName, capacity.Value(), volSizeBytes)
```

Comment:
so can we add logic such that we don't do this accounting for SPs that we have already rejected. Put this at the very end of this function and when you get the spName from the PVC, check if its still in the list using a map and if its not just move on. If you see updateSPCapacityUsage its actually running a loop over the sp list to find the sp.

Reply:
Made requested changes.